File: core/src/test/java/javabot/dao/ChannelDaoTest.java
package javabot.dao;

import java.util.List;
import javax.inject.Inject;

import javabot.model.Activity;
import javabot.model.Channel;
import org.testng.Assert;
import org.testng.annotations.Test;

public class ChannelDaoTest extends BaseServiceTest {
  @Inject
  private ChannelDao channelDao;

  @Test
  public void addChannel() {
    final Channel channel = new Channel();
    final String name = "##" + System.currentTimeMillis();
    channel.setName(name);
    channel.setLogged(true);
    channelDao.save(channel);
    Assert.assertNotNull(channelDao.get(name));
  }

  @Test
  public void getChannels() {
    final List channels = channelDao.getChannels();
    Assert.assertTrue(!channels.isEmpty());

  }

  @Test
  public void currentChannels() {
    final List channels = channelDao.configuredChannels();
    Assert.assertTrue(!channels.isEmpty());
  }

  @Test(enabled=false)
  public void stats() {
    final List<Activity> list = channelDao.getStatistics();
    Assert.assertTrue(!list.isEmpty());
    final Activity activity = list.get(0);
    Assert.assertNotSame(activity.getTotal(), 0);
    Assert.assertNotNull(activity.getPercent());
  }
}

File: src/test/kotlin/javabot/qtest/dao/ChannelDaoTest.kt
package javabot.qtest.dao

import dev.morphia.DeleteOptions
import io.quarkus.test.junit.QuarkusTest
import javabot.dao.BaseServiceTest
import javabot.model.Channel
import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Assertions.assertNotSame
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Disabled
import org.junit.jupiter.api.Test

@QuarkusTest
class ChannelDaoTest : BaseServiceTest() {
    @Test
    fun addChannel() {
        val channel = Channel()
        val name = "##" + System.currentTimeMillis()
        channel.name = name
        channel.logged = true
        channelDao.save(channel)
        assertNotNull(channelDao.get(name))
    }

    @Test
    fun getChannels() {
        assertTrue(!channelDao.getChannels().isEmpty())
    }

    @Test
    fun currentChannels() {
        assertTrue(!channelDao.configuredChannels().isEmpty())
    }

    @Test
    @Disabled
    fun stats() {
        datastore.find(Channel::class.java).delete(DeleteOptions().multi(true))
        val list = channelDao.getStatistics()
        assertTrue(list.isNotEmpty())
        val activity = list[0]
        assertNotSame(0, activity.total)
        assertNotNull(activity.getPercent())
    }
}
